### PR TITLE
Allow adding self-resolving dependencies directly to Test Suites

### PR DESCRIPTION
**Pull request by tresat**
Adds `localGroovy`, `gradleApi` and `gradleTestKit`

### CODE REVIEW

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java [hunk 1]
```diff
@@ -116,4 +116,28 @@ public interface JvmComponentDependencies {
...
+    /**
+     * Creates a dependency on the <a href="https://docs.gradle.org/current/userguide/test_kit.html" target="_top">Gradle test-kit</a> API.
+     *
+     * @return The dependency.
+     * @since 7.6
+     */
+    Dependency gradleTestKit();
+
+    /**
+     * Creates a dependency on the Groovy that is distributed with the current version of Gradle.
```
Comment (jvandort) [inline subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java:137]: `Groovy version`? `Groovy implementation`?

**Review by jvandort — APPROVED** [review]

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java (hunk 1 above)
Comment (tresat) [reply subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java:137]: Have to use the GAV for that (and should probably prefer it in most cases anyway).  This method doesn't take a version even at the top-level - the version you get depends on the version of Gradle you're running.

Comment (octylFractal) [reply subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java:137]: I think this was a question about the phrasing in the Javadoc, you say "the Groovy"  and it sounds like there should be another word there.

Comment (jvandort) [reply subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java:137]: Yeah Octavia is right. I should have used more words here as well

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**